skbio/io/format/stockholm.py

```diff
@@ -478,9 +478,13 @@ def add_gf_metadata(self, feature_name, feature_data):
         # Handles extra line(s) of an already created tree
         elif feature_name == 'NH' and feature_name in self._metadata:
             trees = self._metadata[feature_name]
-            tree_id = list(trees.keys())[-1]
-            self._metadata[feature_name][tree_id] = (trees[tree_id] +
-                                                     feature_data)
+            if isinstance(trees, dict):
+                tree_id = list(trees.keys())[-1]
```

Comment:
Getting the most recently inserted key in an `OrderedDict` can be performed in constant time like this (based on this [SO question](http://stackoverflow.com/questions/9917178/last-element-in-ordereddict)):

```python
tree_id = next(reversed(trees))
```

Sorry for not catching this earlier in review, probably worth updating in this PR since it's a small change.